crates/sui-core/src/authority.rs

```diff
@@ -2147,6 +2147,12 @@ impl AuthorityState {
         )?;
 
         // Create and use a dummy gas object if there is no gas object provided.
+        assert!(
+            DEV_INSPECT_GAS_COIN_VALUE >= max_tx_gas,
+            "DEV_INSPECT_GAS_COIN_VALUE {} cannot be less than max_tx_gas {}",
+            DEV_INSPECT_GAS_COIN_VALUE,
+            max_tx_gas
+        );
```

Comment:
@mwtian what is the intent of this `assert`?
I keep going back and forth on whether or not I would want it there.
Thoughts?

Reply:
In case we need to update DEV_INSPECT_GAS_COIN_VALUE or max_tx_gas again, I hope we can say a few minutes of debugging when this is violated. I'm ok with removing the assert. Another alternative is to remove DEV_INSPECT_GAS_COIN_VALUE and use max_tx_gas instead. What do you think?

Reply:
> Another alternative is to remove `DEV_INSPECT_GAS_COIN_VALUE` and use max_tx_gas instead. What do you think?

I thought of that too and `DEV_INSPECT_GAS_COIN_VALUE` in `authority.rs` feels really core... 
We should try to remove that (or move it in test land) and see that it works but it feels we can do that as a subsequent PR too?